src/composables/useViewState.js

```diff
@@ -0,0 +1,29 @@
+import { ref } from 'vue'
+
+// optionsArray ref needs to be defined outside of export function to ensure
+// state persists between workflows
+
+const optionsArray = ref([])
+
+export default function useViewState () {
+  const options = {
+    tree: {
+      tasksFilter: { type: Object, value: { id: null, states: null } },
+      expandAll: { type: Array, value: [] },
```

Comment:
Ideally the definition of these options would lie in the components they come from rather than this compostable.

We can think of views as cylc-ui plugins and this compostable as part of the support infrastructure for said plugins. The code infrastructure shouldn't need to have any plugin logic hardcoded within it.